connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/ErrorFactoryBuilder.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.client
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+private[client] class ErrorFactoryBuilder {
+
+  private val constructors = mutable.Map.empty[String, (String, Throwable) => Throwable]
+
+  /**
+   * registerConstructors register throwableCtr in ErrorFactoryBuilder.
+   * @param throwableCtr the constructor that construct Throwable based on message and cause.
+   * @return this
+   */
+  def registerConstructor[T <: Throwable: ClassTag](
```

Comment:
Can we just move this function into `GrpcExceptionConverter`, and just use a map? One example is `FunctionRegistry.expressions`. I really don't think this will be reused somewhere, and we can factor it out if it's going to be reused in the future.

If you have a plan to improve it very soon and you believe this class will be more complicated/reused, I am fine. If not for now, I wouldn't make it as a separate class for now.